graphql_compiler/compiler/ir_lowering_match/__init__.py

```diff
@@ -96,6 +97,8 @@ def lower_ir(ir_blocks, query_metadata_table, type_equivalence_hints=None):
     ir_blocks = rewrite_binary_composition_inside_ternary_conditional(ir_blocks)
     ir_blocks = merge_consecutive_filter_clauses(ir_blocks)
     ir_blocks = lower_has_substring_binary_compositions(ir_blocks)
+    ir_blocks = lower_starts_with_binary_compositions(ir_blocks)
+    ir_blocks = lower_ends_with_binary_compositions(ir_blocks)
```

Comment:
Since the three functions on lines 99-101 have similar behavior and implementations, what do you think about merging them? Would that increase the complexity too much, or might it be okay?

Reply:
I think all your suggestions should be resolved in my most recent commit - lmk what you think